src/transformers/configuration_utils.py

```diff
@@ -791,8 +791,8 @@ def to_dict(self) -> Dict[str, Any]:
             `Dict[str, Any]`: Dictionary of all the attributes that make up this configuration instance.
         """
         output = copy.deepcopy(self.__dict__)
-        if hasattr(self.__class__, "model_type"):
-            output["model_type"] = self.__class__.model_type
+        if hasattr(self, "model_type"):
+            output["model_type"] = self.model_type
```

Comment:
To save the actual `model_type`. 

I agree that allowing any specified model type (not the class attribute one) opens a lot of doors for errors.
But how we are going to deal with `decision_transformer_gpt2` and `DecisionTransformerGPT2Model`?

Maybe a new config class `DecisionTransformerGPT2Config`?